### DISCUSSION TITLE
The new version does not install correctly

### DISCUSSION THREAD

**Discussion by 10100010010001001010** [axmolengine/axmol#1274] [General]
![image](https://github.com/axmolengine/axmol/assets/120378517/ba39818c-14cd-43cb-8119-e11f7c305fdd)

that is what i get after execute the setup file via powershell. Then i trying to run axmol, but it gives the error "is not an internal or external command, executable program, or batch file" in cmd and similar error in the powershell

**Comment by halx99** [comment]
Needs restart powershell console terminal

**Reply by 10100010010001001010** [reply]
![image](https://github.com/axmolengine/axmol/assets/120378517/483febd2-b0d9-4ae1-b76b-b266c9d3e4a5)

I have done this many times and reinstalled several times

**Reply by halx99** [reply]
what's your host os? windows, linux or macos?

**Reply by 10100010010001001010** [reply]
my os is windows

**Reply by 10100010010001001010** [reply]
the problem was solved by manually adding the **axmol** and **console/bin** folders. If I remember correctly, for some reason only the path to the main folder was added, and both were not added to the **path** environment variable

**Reply by halx99** [reply]
https://github.com/axmolengine/axmol/blob/main/setup.ps1#L28 , the windows use `[Environment]::SetEnvironmentVariable('PATH', $PATH, 'User')` to save PATH, should works

**Reply by rh101** [reply]
@halx99 I just tested this (Windows 10), and it's not adding the path to the user $PATH variable.  I deleted all the existing entries, opened a new powershell window, and ran the `setup.ps1`.  It only created the `AX_ROOT` entry, and nothing added to the either the user or system path variables.

**Reply by halx99** [reply]
ok, will fix later

**Reply by rh101** [reply]
These changes work:

    $pathList = [System.Collections.ArrayList]$env:PATH.Split(';')
    if ($pathList.IndexOf($AX_CONSOLE_BIN) -eq -1) {
        $pathList = [System.Collections.ArrayList][Environment]::GetEnvironmentVariable('PATH', 'User').Split(';')
        $pathList.Insert(0, $AX_CONSOLE_BIN)
        $PATH = $pathList -join ';'
        [Environment]::SetEnvironmentVariable('PATH', $PATH, 'User')
    }

The `if` condition was inverted, and the `[Environment]::GetEnvironmentVariable` was returning an immutable list which needs to be converted before insert will work.

@halx99 Do you want to open a PR with the changes?

**Reply by halx99** [reply]
Glad to you can post a PR

**Comment by antn9x** [comment]
It's a bug. You need add Path manual to `axmol/tools/console/bin` folder.

**Comment by antn9x** [comment]
still have bug on my mac intel chip. cannot setup 'axmol' command.

**Reply by rh101** [reply]
You should create an issue in the tracker specific to the problem you're having.

**Reply by antn9x** [reply]
ok, i am test more and will create detailed issue.
